client/state/sites/features/actions.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+
+import debugFactory from 'debug';
+import i18n from 'i18n-calypso';
+
+const debug = debugFactory( 'calypso:site-features:actions' );
+
+/**
+ * Internal dependencies
+ */
+import { createSiteFeaturesObject } from './assembler';
+import {
+	SITE_FEATURES_FETCH,
+	SITE_FEATURES_FETCH_COMPLETED,
+	SITE_FEATURES_FETCH_FAILED,
+	SITE_FEATURES_REMOVE,
+} from 'calypso/state/action-types';
+import wpcom from 'calypso/lib/wp';
+
+/**
+ * Returns an action object to be used in signalling that features for the given site have been cleared.
+ *
+ * @param {number} siteId identifier of the site
+ * @returns {object} the corresponding action object
+ */
+export function clearSiteFeatures( siteId ) {
+	return {
+		type: SITE_FEATURES_REMOVE,
+		siteId,
+	};
+}
+
+/**
+ * Fetches features for the given site.
+ *
+ * @param {number} siteId identifier of the site
+ * @returns {Function} a promise that will resolve once fetching is completed
+ */
+export function fetchSiteFeatures( siteId ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: SITE_FEATURES_FETCH,
+			siteId,
+		} );
+
+		return new Promise( ( resolve ) => {
+			wpcom.undocumented().getSiteFeatures( siteId, ( error, data ) => {
+				if ( error ) {
+					debug( 'Fetching site features failed: ', error );
+
+					const errorMessage =
+						error.message ||
+						i18n.translate(
+							'There was a problem fetching site features. Please try again later or contact support.'
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'There was a problem fetching site products. Please try again later or contact support.' )` *ES Score: 7*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/yoast-premium-allow-setting-primary-domain)